static/src/js/app/components/quick-add-button/QuickAddButton.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div class="quick-add">
+    <Modal
+      ref="Modal"
+      title="Add Word to Personal Vocabulary List"
+      description="Modal to quick add vocab"
+      ><QuickAddVocabForm ref="QuickAddVocabForm"
+    /></Modal>
+    <button type="button" class="btn btn-primary" @click="openModal">
+      Quick Add Vocab
+    </button>
+  </div>
+</template>
+<script>
+  import Modal from '../Modal.vue';
+  import QuickAddVocabForm from './QuickAddVocabForm.vue';
+  import { FETCH_ME } from '../../constants';
+
+  export default {
+    components: { Modal, QuickAddVocabForm },
+    async created() {
+      await this.$store.dispatch(FETCH_ME);
```

Comment:
I'm wondering if it would make sense to have the `FETCH_ME` action dispatched from the `Dashboard.vue` component instead of here. My assumption is that the "me" data only needs to be read once, when the dashboard is loaded. I could imagine other sub-components on the dashboard needing that data, and we wouldn't want every sub-component dispatching the same action. So if the dashboard managed that state by dispatching the action, then sub-components wouldn't need to manage it, they could just read the state directly from the store. I could be thinking about this wrong, so curious what you think.

Reply:
I think due to how hedera loads in different vue components periodically some of the vuex state might not persist between page navigation. Since we are planning to add this button site wide, might need to be updated in [ATGU-3160](https://jira.huit.harvard.edu/browse/ATGU-3160)

Reply:
Yeah, it may require further refactoring when we get to that story.